File: PackageTrackingAPI/Models/Package.cs
namespace PackageTrackingAPI.Models
{
    public class Package
    {
        public int Id { get; set; }
        public string TrackingNumber { get; set; } = string.Empty;
        
        // Sender Information
        public string SenderName { get; set; } = string.Empty;
        public string SenderAddress { get; set; } = string.Empty;
        public string SenderPhone { get; set; } = string.Empty;
        
        // Recipient Information
        public string RecipientName { get; set; } = string.Empty;
        public string RecipientAddress { get; set; } = string.Empty;
        public string RecipientPhone { get; set; } = string.Empty;
        
        // Status Information
        public PackageStatus CurrentStatus { get; set; }
        public DateTime CreatedDate { get; set; }
        public DateTime LastUpdated { get; set; }
        
        // Navigation property for status history
        public List<StatusHistory> StatusHistories { get; set; } = new List<StatusHistory>();
    }
}

File: package-tracking-frontend/src/types/package.ts
export enum PackageStatus {
  Created = 1,
  Sent = 2,
  Accepted = 3,
  Returned = 4,
  Canceled = 5,
}

export interface StatusHistory {
  id: number;
  status: PackageStatus;
  timestamp: string;
}

export interface Package {
  id: number;
  trackingNumber: string;
  senderName: string;
  senderAddress: string;
  senderPhone: string;
  recipientName: string;
  recipientAddress: string;
  recipientPhone: string;
  currentStatus: PackageStatus;
  createdAt: string;
  statusHistory: StatusHistory[];
}

export interface CreatePackageRequest {
  senderName: string;
  senderAddress: string;
  senderPhone: string;
  recipientName: string;
  recipientAddress: string;
  recipientPhone: string;
}

export interface UpdateStatusRequest {
  newStatus: PackageStatus;
}
